remote/remote.go

```diff
@@ -71,50 +98,119 @@ func NewImage(repoName string, keychain authn.Keychain, ops ...ImageOption) (*Im
 		keychain: keychain,
 		repoName: repoName,
 		image:    image,
+		platform: defaultPlatform(),
+	}
+
+	ri, err = processImageOptions(ri, ops)
+	if err != nil {
+		return nil, err
 	}
 
+	ri, err = prepareImage(ri)
+	if err != nil {
+		return nil, err
+	}
+
+	return ri, nil
+}
+
+func processImageOptions(image *Image, ops []interface{}) (*Image, error) {
+	sort.Slice(ops, func(i, _ int) bool {
+		switch ops[i].(type) {
+		case InitialImageOption:
+			return true
+		default:
+			return false
+		}
+	})
+
 	for _, op := range ops {
-		ri, err = op(ri)
+		var err error
+
+		switch o := op.(type) {
+		case InitialImageOption:
+			image, err = o(image)
+		case ImageOption:
+			image, err = o(image)
+		}
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	return image, nil
+}
+
+func prepareImage(ri *Image) (*Image, error) {
+	imageConfigFile, err := ri.image.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+
+	if imageConfigFile.OS != "windows" || len(imageConfigFile.RootFS.DiffIDs) != 0 {
+		return ri, nil
+	}
+
+	layerBytes, err := layer.WindowsBaseLayer()
+	if err != nil {
+		return nil, err
+	}
+	windowsBaseLayer, err := tarball.LayerFromReader(layerBytes)
+	if err != nil {
+		return nil, err
+	}
+	ri.image, err = mutate.AppendLayers(ri.image, windowsBaseLayer)
+	if err != nil {
+		return nil, err
+	}
+
 	return ri, nil
 }
 
-func newV1Image(keychain authn.Keychain, repoName string) (v1.Image, error) {
+func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platform) (v1.Image, error) {
 	ref, auth, err := referenceForRepoName(keychain, repoName)
 	if err != nil {
 		return nil, err
 	}
 
-	image, err := remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport))
+	image, err := remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1.Platform(platform)))
```

Comment:
Did we ensure that this still works if you are fetching a windows image (not from a manifest list) without `WithPlatform`? I am assuming it does after looking at the GGCR code, but just double checking.

Reply:
Since `descriptor.Image()` (called by `remote.Image()` [currently ignores](https://github.com/google/go-containerregistry/blob/51f01e739161bf35d663240a8ca404777d7d9378/pkg/v1/remote/descriptor.go#L152-L156) the platform for any non-manifest-list result. 

I think the [first remote/#NewImage/#BaseImage](https://github.com/buildpacks/imgutil/blob/038872784f71cf5d70fce08b92ebf5716337180d/remote/remote_test.go#L143) test kinda covers this, since it doesn't match ggcrs `linux/amd64` default platform. I can change that test, or add another that deviates from all the default platform fields.

Reply:
Added new case: https://github.com/buildpacks/imgutil/pull/91/files#diff-55e016054597444d1031ab5c620b77bdf8118f0ee1f82888a2b851850e151726R171